test/inline_test_with_deps/test_with_srcs/BUILD

```diff
@@ -0,0 +1,20 @@
+package(default_visibility = ["//visibility:public"])
+
+load(
+    "//rust:rust.bzl",
+    "rust_library",
+    "rust_test",
+)
+
+rust_library(
+    name = "inline",
+    edition = "2018",
+    srcs = ["src/lib.rs"],
+)
+
+rust_test(
+    name = "inline_test",
+    crate = ":inline",
+    srcs = ["src/extra.rs"],
```

Comment:
What is the motivation for doing this without making it a separate test target?

Reply:
primarily to hold out things like `test_ds_helpers.rs` that are test helpers used in inline tests but are not really general code for the resulting library.

I could also go either way